.github/workflows/benchmark.yaml

```diff
@@ -0,0 +1,46 @@
+name: performance
+
+on:
+  pull_request:
+    types: [opened, reopened]
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - uses: actions/setup-go@v5
+      with:
+        go-version-file: go.mod
+    
+    - name: Install dependencies
+      run: go get .
+
+    - name: Run golang benchmarks
+      run: go test -run="^$" -bench=. ./internal/...
+
+    - name: Install benchstat
+      run: go install golang.org/x/perf/cmd/benchstat@latest
+
+    - name: Set Golang binaries path
+      run: echo "`go env GOPATH`/bin" >> $GITHUB_PATH
+
+    - name: Download previous benchmarks
+      id: download
+      uses: actions/download-artifact@v4
+      with:
+        name: benchmark-results
+        path: stats/benchmark.txt
+      continue-on-error: true
+
+    - name: Compare benchmark results
+      if: steps.download.outcome == 'success'
+      run: benchstat benchmark.txt stats/benchmark.txt
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: benchmark-results
+        path: benchmark.txt
+        overwrite: true
```

Comment:
Still needs a step to compare the change in performance as a result of the PR .

In addition, what do we think would be a good threshold for a computing performance degradation